lib/model/query/entities.js

```diff
@@ -118,6 +118,16 @@ const getDefBySubmissionDefId = (submissionDefId) => ({ maybeOne }) =>
   maybeOne(sql`select * from entity_defs where "submissionDefId" = ${submissionDefId} limit 1`)
     .then(map(construct(Entity.Def)));
 
+// This will check for an entity related to any def of the same submission
+// as the one specified. Used when trying to reapprove an edited submission.
```

Comment:
Is this necessary? It looks expensive, especially once we have long-lived entities with a rich history. Any chance we could instead do this when receiving the submission? If it's an edit, don't consider it for entity processing?

Reply:
Yeah, I didn't feel great about this query.

Looking at the event data, we have the submission ID, too, so I'm adding a commit that uses that instead. That's more of the intent, anyway, that we'll only allow one entity per submission.

Reply:
> If it's an edit, don't consider it for entity processing?

If approval comes after the edit, then we should use the latest submission version as the basis of the entity, right?

> Any chance we could instead do this when receiving the submission?

The event that's being hooked into isn't the request, it's the `submission.update` audit log action. After each change to a submission's review state, a worker will check whether an entity needs to be created from the submission.

Reply:
> If approval comes after the edit, then we should use the latest submission version as the basis of the entity, right?

Yes.

> The event that's being hooked into isn't the request

Right, I'm getting ahead of myself here and thinking of the create on receive flow.

It's good to see some simplification and it does seem like that's probably as good as it gets.